is_core/generic_views/detail_views.py

```diff
@@ -139,11 +145,48 @@ def get_fieldsets(self):
             self.core.get_fieldsets(self.request)
         )
 
-    def generate_fieldsets(self):
+    def generate_fieldsets(self, obj):
         fieldsets = self.get_fieldsets()
+
         if fieldsets is None:
-            fieldsets = [(None, {'fields': self.get_fields() or ()})]
-        return fieldsets
+            fieldsets = [
+                (None, {
+                    'fields': self.get_fields() or []
+                })
+            ]
+            for inline_view in self.get_inline_views() or ():
+                inline_view_inst = (
+                    inline_view(self.request, self, obj) if isinstance(inline_view, type) else inline_view
+                )
+                # Only inline view that can be rendered is added to formset
+                fieldsets.append((
+                    inline_view_inst.get_title(), {
+                        'inline_view': inline_view,
+                    }
+                ))
```

Comment:
is this supposed to be here or only the one inside the if? the comment suggests that it should be only added if it can be rendered which is inside the if

Reply:
yes that's a goblin :)